sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -31,11 +31,22 @@ object InterpretedPredicate {
   def create(expression: Expression, inputSchema: Seq[Attribute]): InterpretedPredicate =
     create(BindReferences.bindReference(expression, inputSchema))
 
-  def create(expression: Expression): InterpretedPredicate = new InterpretedPredicate(expression)
+  def create(expression: Expression): InterpretedPredicate = {
+    val predicate = new InterpretedPredicate(expression)
+    predicate.initialize(0)
```

Comment:
`0` is not a `partitionIndex` and  `0` will go to `n.initialize(0)` in line 46. Is it correct?